### ISSUE TITLE
Extract the lib to the rb-lib

### ISSUE THREAD

**Issue by Kouprin** [near/rainbow-bridge#378]
The purpose is to extract everything that is common to all Token Connectors to the new rb-lib from https://github.com/near/rainbow-bridge-frontend/blob/master/src/js/transfers.js and update rb-frontend to use the library.

**Comment by chadoh** [comment]
I'm not sure if this is actually blocked by #373 

@frol I am thinking I'll name this library "rainbow-bridge-browser" for now. If we find a way to make have a version that works in Node as well, then we can rename it to "rainbow-bridge-lib". If we find ways to split it up and use a pluggable architecture for keeping various proof logic in isolated libraries, we can also rename the core library to something like `@rainbow-bridge/core` on NPM. Let me know if you have other ideas.

**Comment by Kouprin** [comment]
> I'm not sure if this is actually blocked by #373

Yeah, you are right, removed it from the topic.

According to my understanding to how https://github.com/near/rainbow-token-connector is implemented, the interface of other Token Connectors should be the same. Can we have generic "rainbow-bridge-browser" instead of strong connectivity ERC-20/NEP-21?

**Comment by chadoh** [comment]
Right now, the logic for creating a proof is stored [in the frontend code](https://github.com/near/rainbow-bridge-frontend/blob/7cec7c4c48422bc21f3738cad1ea5b0bcc2fd865/src/js/transfers.js#L353-L384). I think we should keep this as an option, even if we provide some centralized endpoints where proofs can be generated via an API call (and the option to self-host such an API service).

My understanding is that the exact shape of these proofs will change for each sort of connector. If that's true, then I think providing a pluggable interface seems nice to me. Many projects might end up with dependencies like:

```json
"dependencies": {
  "@ethXnear/core": "^1.0.0",
  "@ethXnear/connector-fungible": "^1.0.0",
  "@ethXnear/connector-nonfungible": "^1.0.0"
}
```

But for projects that need their own custom connector, they could copy one of our `connector-*` libraries and publish their own library, to be used in their frontend:

```json
"dependencies": {
  "@ethXnear/core": "^1.0.0",
  "@ethXnear/connector-nonfungible": "^1.0.0"
  "custom-fungible-connector": "^1.0.0",
}
```

**Comment by djsatok** [comment]
Outdated, see here: https://github.com/near/rainbow-bridge-frontend/pull/75
